src/lease.rs

```diff
@@ -32,6 +34,23 @@ impl Lease {
         self.local_guard = Some(guard);
         self
     }
+
+    /// Asynchronously releases the underlying lock.
+    pub async fn release(mut self) {
+        let client = self.client.clone();
+        let key_lease_v = self.key_lease_v.clone();
+
+        // Drop local guard *before* deleting lease to avoid unfair local acquire advantage.
+        // Dropping the local_guard after deleting would be more efficient however during
+        // contention that efficiency could starve remote attempts to acquire the lease.
+        drop(self.local_guard.take());
+        client.try_clean_local_lock(key_lease_v.0.clone());
+
+        let lease_v = key_lease_v.1.lock().await;
+        let key = key_lease_v.0.clone();
+        // TODO retries, logs?
+        let _ = client.delete_lease(key, *lease_v).await;
+    }
```

Comment:
We can return a result here allowing users to handle `delete_lease` failures in a way drop cannot. (Drop can continue to ignore the error).

```suggestion
    pub async fn release(mut self) -> anyhow::Result<()> {
        let client = self.client.clone();
        let key_lease_v = self.key_lease_v.clone();

        // Drop local guard *before* deleting lease to avoid unfair local acquire advantage.
        // Dropping the local_guard after deleting would be more efficient however during
        // contention that efficiency could starve remote attempts to acquire the lease.
        drop(self.local_guard.take());
        client.try_clean_local_lock(key_lease_v.0.clone());

        let lease_v = key_lease_v.1.lock().await;
        let key = key_lease_v.0.clone();
        client.delete_lease(key, *lease_v).await?;
        Ok(())
    }
```

Reply:
We should also document the result a little, I can add that later though